Fitness/Views/Cells/HabitTrackerOnboardingCell.swift

```diff
@@ -0,0 +1,246 @@
+        trashButton = UIButton()
+        trashButton.setImage(UIImage(named: "trash-icon"), for: .normal)
+        trashButton.addTarget(self, action: #selector(deleteCell), for: .touchUpInside)
+        contentView.addSubview(trashButton)
+        
+        editButton = UIButton()
+        editButton.setImage(UIImage(named: "edit-icon"), for: .normal)
+        editButton.addTarget(self, action: #selector(edit), for: .touchUpInside)
+        contentView.addSubview(editButton)
+        
+        pinButton = UIButton()
+        pinButton.setImage(UIImage(named: "pin-icon"), for: .normal)
+        pinButton.addTarget(self, action: #selector(pin), for: .touchUpInside)
+        contentView.addSubview(pinButton)
+        
+        clearTextButton = UIButton()
+        clearTextButton.setImage(UIImage(named: "keyboard-cancel"), for: .normal)
+        clearTextButton.addTarget(self, action: #selector(clearText), for: .touchUpInside)
+        contentView.addSubview(clearTextButton)
+        
+        isSwiped = false
+        canEdit = false
+        editingTitle = false
+        trashButton.isHidden = true
+        editButton.isHidden = true
+        pinButton.isHidden = true
+        clearTextButton.isHidden = true
+        
+        setupConstraints()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - CONSTRAINTS
+    func setupConstraints() {
+        let buttonSize = 30
+        
+        dotsImage.snp.makeConstraints { make in
+            make.leading.equalToSuperview().offset(35)
+            make.width.equalTo(8)
+            make.height.equalTo(14)
+            make.centerY.equalToSuperview()
+        }
+        
+        titleLabel.snp.makeConstraints { make in
+            make.leading.equalTo(dotsImage.snp.trailing).offset(10)
+            make.height.equalTo(14)
+            make.centerY.equalToSuperview()
+            make.trailing.equalToSuperview().offset(-21)
+        }
+        
+        pinButton.snp.makeConstraints { make in
+            make.width.height.equalTo(buttonSize)
+            make.trailing.equalToSuperview().offset(-21)
+            make.centerY.equalToSuperview()
+        }
+        
+        editButton.snp.makeConstraints { make in
+            make.width.height.equalTo(buttonSize)
+            make.trailing.equalTo(pinButton.snp.leading).offset(-6)
+            make.centerY.equalToSuperview()
+        }
+        
+        trashButton.snp.makeConstraints { make in
+            make.width.height.equalTo(buttonSize)
+            make.trailing.equalTo(editButton.snp.leading).offset(-6)
+            make.centerY.equalToSuperview()
+        }
+        
+        clearTextButton.snp.makeConstraints { make in
+            make.height.width.equalTo(16)
+            make.trailing.equalToSuperview().offset(-24)
+            make.centerY.equalToSuperview()
+        }
+    }
+    
+    func updateConstraintsLeftSwipe() {
```

Comment:
Haven't seen this in phone app –– maybe there should be some UIAnimation for the update for swiping?

Reply:
True I didn't consider that, will add!